node/grpc/server_test.go

```diff
@@ -184,7 +184,26 @@ func newTestServerWithConfig(t *testing.T, mockValidator bool, config *node.Conf
 		Validator:      val,
 		ValidationPool: workerpool.New(1),
 	}
-	return grpc.NewServer(config, node, logger, ratelimiter, version.DefaultVersion())
+
+	// Create listeners with OS-allocated ports for testing
+	v1DispersalListener, err := net.Listen("tcp", "0.0.0.0:0")
+	if err != nil {
+		panic("failed to create v1 dispersal listener")
```

Comment:
My personal preference for failing a test is `require.NoError(t, err)` instead of panic.

Reply:
https://github.com/Layr-Labs/eigenda/pull/2109/commits/53e0f6617289de66c5a0e37b7e4fecc09473904b